projects/pastanaga-angular/src/lib/side-nav/side-nav.component.ts

```diff
@@ -37,46 +38,52 @@ export class SideNavComponent implements OnInit, AfterContentInit {
         return !!this._visible;
     }
     set visible(value: boolean) {
-        this._visible = coerceBooleanProperty(value);
+        if (this._mode !== 'desktop' && !this.modeChanged) {
+            this.triggerAnimation(value);
+        } else {
+            this.modeChanged = false;
+            this._visible = coerceBooleanProperty(value);
+        }
+    }
+    @Input()
+    get mode(): string {
+        return this._mode;
     }
-    @Input() toggle?: Subject<boolean>;
+    set mode(value: string) {
+        this.modeChanged = value !== this._mode;
+        this._mode = value;
+    }
+    @Output() close: Subject<void> = new Subject<void>();
     @ViewChild('navBar', { read: ElementRef }) navBar?: ElementRef;
     @ViewChild('tabletOverlay', { read: ElementRef }) tabletOverlay?: ElementRef;
     @ContentChildren(SideNavItemComponent, { descendants: true }) contentChild!: QueryList<SideNavItemComponent>;
     _inverted = false;
-    _visible = false;
-    _mode?: string;
+    _visible = true;
+    _mode = 'desktop';
+    // we need to avoid animation when mode changed (from desktop to tablet for example)
```

Comment:
do we?

Reply:
yes, because we don't want to trigger the animation when you change from desktop to tablet, we only want to trigger it when someone opens or closes the sidenav while the mode is tablet/mobile.